app/queries/subscriber_list_query.rb

```diff
@@ -42,9 +45,11 @@ def find_exact_match_with(query_hash, document_type)
   end
 
   def subscriber_lists_with_key(key)
-    # This uses the `?` hstore operator, which returns true only if the hstore
+    # This uses the `ANY` array operator, which returns true only if the field's keys
     # contains the specified key.
-    SubscriberList.where("#{@query_field} ? :key", key: key)
+    SubscriberList.where(":key = ANY(ARRAY(SELECT json_object_keys(#{@query_field})))",
+      key: key,
+    )
```

Comment:
Looks like it needs to be `SubscriberList.where("(#{@query_field} -> :key) IS NOT NULL", key: key)` but yeah that should work.